plugins/conventional-commits/src/index.ts

```diff
@@ -135,6 +135,56 @@ export default class ConventionalCommitsPlugin implements IPlugin {
       return this.storedGetBump(message);
     };
 
+    auto.hooks.prCheck.tapPromise(this.name, async ({ pr }) => {
+      if (!auto.git) {
+        return;
+      }
+
+      const VERSIONS = [
+        SEMVER.major,
+        SEMVER.minor,
+        SEMVER.patch,
+        "skip",
+      ] as const;
+
+      const labels = await auto.git.getLabels(pr.number);
+
+      // check if semver label is already on PR
+      if (labels.filter((l) => VERSIONS.includes(l as any)).length > 0) {
+        return;
+      }
+
+      const commits = await auto.git?.getCommitsForPR(pr.number);
+
+      const bumps = await Promise.all(
+        commits.map(async (commit) => {
+          try {
+            return await getBump(commit.commit.message);
+          } catch (error) {
+            auto.logger.verbose.info(
+              `No conventional commit message found for ${commit.sha}`
+            );
+          }
+        })
+      );
+
+      const sorted = bumps.sort((bump1, bump2) => {
```

Comment:
i tried using `filter` to remove the undefined by TS still complained that `bump1`/`bump2` could be undefined

Reply:
To get filter to work like that you need to use `is`

https://github.com/intuit/auto/blob/main/packages/core/src/auto.ts#L2157